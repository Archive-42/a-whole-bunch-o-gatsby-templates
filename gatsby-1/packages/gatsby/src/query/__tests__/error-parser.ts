import errorParser from "../error-parser"

describe(`query-error-parser`, () => {
  it.each([
    [
      `specific one`,
      `Variable "Foo" of required type "Bar" was not provided.`,
      `85920`,
    ],
    [`totally vague one`, `foo bar`, `85901`],
    [
      `real error`,
      `this error should show a trace`,
      `85901`,
      new Error(`this error should show a trace`),
    ],
  ])(`%s`, (_name, message, expectedId, error = undefined) => {
    const structured = errorParser({
      message,
      filePath: `test.js`,
      location: { start: { line: 5, column: 10 } },
      error: error instanceof Error ? error : undefined,
    })
    expect(structured).toMatchSnapshot()
    expect(structured.id).toEqual(expectedId)
  })
})
