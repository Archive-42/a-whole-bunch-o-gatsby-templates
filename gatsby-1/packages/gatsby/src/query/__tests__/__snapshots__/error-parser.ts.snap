// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query-error-parser real error 1`] = `
Object {
  "context": Object {
    "sourceMessage": "this error should show a trace",
  },
  "error": [Error: this error should show a trace],
  "filePath": "test.js",
  "id": "85901",
  "location": Object {
    "start": Object {
      "column": 10,
      "line": 5,
    },
  },
}
`;

exports[`query-error-parser specific one 1`] = `
Object {
  "context": Object {
    "sourceMessage": "Variable \\"Foo\\" of required type \\"Bar\\" was not provided.",
    "variableName": "Foo",
    "variableType": "Bar",
  },
  "filePath": "test.js",
  "id": "85920",
  "location": Object {
    "start": Object {
      "column": 10,
      "line": 5,
    },
  },
}
`;

exports[`query-error-parser totally vague one 1`] = `
Object {
  "context": Object {
    "sourceMessage": "foo bar",
  },
  "filePath": "test.js",
  "id": "85901",
  "location": Object {
    "start": Object {
      "column": 10,
      "line": 5,
    },
  },
}
`;
